addons/account/tests/test_chart_template.py

```diff
@@ -355,17 +360,27 @@ def local_get_data(self, template_code):
         self.assertEqual(len(taxes_1_companies), 4)
 
     def test_update_account_codes_conflict(self):
-        # Change code of an existing account to something else
+        # Change code of existing accounts to something else
         standard_account = self.env['account.chart.template'].ref('test_account_income_template')
         standard_account.code = '111111'
 
+        standard_account_b = self.env['account.chart.template'].ref('test_account_code_template')
+        standard_account_b.code = '123'
+
         # create a new account with the same code as an existing one
         problematic_account = self.env['account.account'].create({
             'code': '222221',
             'name': 'problematic_account',
             'company_id': self.company_1.id,
         })
 
+        # create an account with a normalized existing code
+        problematic_account_b = self.env['account.account'].create({
+            'code': '123000',
```

Comment:
But we wouldn't find the account with `123` without the `000` anymore.

Reply:
Indeed, I was focused on the duplicates and I hadn't considered that case.